### ISSUE TITLE
VAR & LM

### ISSUE THREAD

**Issue by sammer21** [Kydoimos97/CapstoneMSBA2020#40]
I have found a couple guides to implementing VAR and LM models on time series data in python. 

[Here is VAR](https://www.analyticsvidhya.com/blog/2018/09/multivariate-time-series-guide-forecasting-modeling-python-codes/).

[Here is LM.](https://www.kaggle.com/antoinekrajnc/simplest-time-series-using-linear-regression)

**Comment by sammer21** [comment]
`1  VAR
df['date'] = pd.to_datetime(df.date)
df['price'] = df['price'].astype(int)
df['sales'] = df['sales'].astype(int)
​
data = df.drop(['date', 'item_id'], axis = 1)
data.index = df.date
train = data[:int(0.8*(len(data)))]
test = data[:int(0.8*(len(data))):]
#fit the model
from statsmodels.tsa.vector_ar.var_model import VAR 
​
model = VAR(endog= train)
model_fit = model.fit()

#make predictions on validation
prediction = model_fit.forecast(model_fit.y, steps = len(test))

import math
from sklearn.metrics import mean_squared_error
cols = data.columns
​
#changing test to an array
​
#converting predictions to dataframe
pred = pd.DataFrame(index = range(0,len(prediction)), columns=[cols])
​
for j in range (0,9):
    for i in range(0, len(prediction)):
        pred.iloc[i][j] = prediction[i][j]
​
#check rmse
for i in cols:
    print('rmse value for', i, 'is : ', math.sqrt(mean_squared_error(pred[i], test[i])))

df.dtypes

#make final predictions
model = VAR(endog=data)
model_fit = model.fit()
yhat = model_fit.forecast(model_fit.y, steps = 1)
print(yhat)
[[8.65900000e+03 1.35182606e+06 5.07127382e+02 4.64203321e+03
  1.50683241e+01 2.63897265e+00 4.05159445e+06 4.28130430e+00
  2.02273186e+00 8.94337854e+00]]


2  LM
import numpy as np 
import pandas as pd 
import os 
df = pd.read_csv("snickers_df.csv")
-
import datetime
df.date = df.date.apply(lambda x:datetime.datetime.strptime(x, "%Y-%m-%d"))
df.info()

for i in range(4,10):
    df["lag_{}".forrmat(i)] = df.sales(i)
print("done")`

**Comment by Kydoimos97** [comment]
Not an area of focus anymore
